### PR TITLE
Issue/238 add cypress tests

### PR DESCRIPTION
**Pull request by msentwickler**
## Related Issue
https://github.com/E-Edu/frontend/issues/238

## Type of change

- [ ] Bug fix
- [x] Feature request
- [ ] Documenting code

## Checklists

- [x] I referenced the according Issue
  - [x] I created an Issue
  - [x] I put a link in the Section [Related Issue](#related-issue)
- [ ] I tested my code with multiple browsers
- [x] My commit messages are good and understandable

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -1,3 +1,5 @@
 {
-    "baseurl": "http://localhost:3000"
+    "baseUrl": "http://localhost:3000",
+    "viewportWidth": 1080,
```
Comment (adrian-goe) [inline cypress.json:3]: 1920x1080 should be the default

File: cypress/fixtures/example.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (adrian-goe) [inline cypress/fixtures/example.json:1]: do we still need you?

File: cypress/integration/DoingTasks.spec.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
+describe('Doing tasks Test', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:1]: lowercase

File: cypress/integration/DoingTasks.spec.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    beforeEach(() => {
+        cy.visit('/dashboard');
+    });
+
+    it('Should open German subject', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:6]: lowercase

File: cypress/integration/DoingTasks.spec.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            .url().should('include', '/task/subject')
+            .get('input[placeholder=test]').type('Answer')
+            .get('.subject-task-next').click()
+            .url().should('include', '/task/result')
+            .visit('/task/subject')
+            .get('.subject-task-skip').click()
+            .url().should('include', '/task/result');
+    });
+
+    it('Should open Computer Sience subject', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:22]: lowercase

File: cypress/integration/DoingTasks.spec.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            .url().should('include', '/task/subject')
+            .get('input[placeholder=test]').type('Answer')
+            .get('.subject-task-next').click()
+            .url().should('include', '/task/result')
+            .visit('/task/subject')
+            .get('.subject-task-skip').click()
+            .url().should('include', '/task/result');
+    });
+
+    it('Should open History subject', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:38]: lowercase

File: cypress/integration/DoingTasks.spec.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            .url().should('include', '/task/subject')
+            .get('input[placeholder=test]').type('Answer')
+            .get('.subject-task-next').click()
+            .url().should('include', '/task/result')
+            .visit('/task/subject')
+            .get('.subject-task-skip').click()
+            .url().should('include', '/task/result');
+    });
+
+    it('Should open Politics subject', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:54]: lowercase

File: cypress/integration/DoingTasks.spec.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            .url().should('include', '/task/subject')
+            .get('input[placeholder=test]').type('Answer')
+            .get('.subject-task-next').click()
+            .url().should('include', '/task/result')
+            .visit('/task/subject')
+            .get('.subject-task-skip').click()
+            .url().should('include', '/task/result');
+    });
+
+    it('Should open Physics subject', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:70]: lowercase

File: cypress/integration/DoingTasks.spec.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+            .url().should('include', '/task/subject')
+            .get('input[placeholder=test]').type('Answer')
+            .get('.subject-task-next').click()
+            .url().should('include', '/task/result')
+            .visit('/task/subject')
+            .get('.subject-task-skip').click()
+            .url().should('include', '/task/result');
+    });
+
+    it('Should open Biology subject', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:86]: lowercase

File: cypress/integration/DoingTasks.spec.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+            .url().should('include', '/task/subject')
+            .get('input[placeholder=test]').type('Answer')
+            .get('.subject-task-next').click()
+            .url().should('include', '/task/result')
+            .visit('/task/subject')
+            .get('.subject-task-skip').click()
+            .url().should('include', '/task/result');
+    });
+
+    it('Should open Chemistry subject', () => {
```
Comment (adrian-goe) [inline cypress/integration/DoingTasks.spec.js:102]: lowercase

File: cypress/integration/Like.spec.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
+describe('Liking test', () => {
```
Comment (adrian-goe) [inline cypress/integration/Like.spec.js:1]: lowercase

File: cypress/integration/Like.spec.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    it('Should give thumb up', () => {
```
Comment (adrian-goe) [inline cypress/integration/Like.spec.js:2]: lowercase

File: cypress/integration/Like.spec.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+        cy.visit('/reports/list')
+            .get('.thumbs-up')
```
Comment (adrian-goe) [inline cypress/integration/Like.spec.js:4]:
```ts            
.get('.thumbs-up').first().click()
```
group together

File: cypress/integration/Like.spec.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+            .first()
+            .click()
+            .get('.thumbs-up-value')
+            .first()
+            .invoke('text')
+            .should('contain', '4');
+    });
+    it('Should give thumb down', () => {
```
Comment (adrian-goe) [inline cypress/integration/Like.spec.js:12]: lowercase

File: cypress/integration/Like.spec.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+            .get('.thumbs-down-value')
+            .first()
+            .invoke('text')
+            .should('contain', '179')
+            .get('.thumbs-up-value')
+            .first()
+            .invoke('text')
+            .should('contain', '3');
+    });
+    it('Should reset thumbs', () => {
```
Comment (adrian-goe) [inline cypress/integration/Like.spec.js:25]: lowercase

File: cypress/integration/Profile.spec.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
+describe('Profile test', () => {
```
Comment (adrian-goe) [inline cypress/integration/Profile.spec.js:1]: lowercase

File: cypress/integration/Profile.spec.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+    it('Should edit profile name', () => {
```
Comment (adrian-goe) [inline cypress/integration/Profile.spec.js:2]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 18]
```diff
@@ -0,0 +1,39 @@
+describe('Sidebar linking test', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:1]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+    it('Should link to tasks', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:2]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+        cy.visit('/dashboard')
+            .get('.menu-element').eq(1).click()
+            .url().should('include', '/task/lecture');
+    });
+    it('Should link to home', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:7]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.menu-element').eq(0).click()
+            .url().should('include', '/dashboard');
+    });
+    it('Should link to new task', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:11]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.menu-element').eq(2).click()
+            .url().should('include', '/dashboard');
+    });
+    it('Should link to my tasks', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:15]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.menu-element').eq(3).click()
+            .url().should('include', '/task/list');
+    });
+    it('Should link to correction', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:19]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.menu-element').eq(4).click()
+            .url().should('include', '/correction/review');
+    });
+    it('Should link to reports', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:23]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.menu-element').eq(5).click()
+            .url().should('include', '/reports/list');
+    });
+    it('Should link to add teacher', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:27]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 26]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.menu-element').eq(6).click()
+            .url().should('include', '/teacher/add');
+    });
+    it('Should link to settings', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:31]: lowercase

File: cypress/integration/Sidebar.spec.js [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.menu-element').eq(7).click()
+            .url().should('include', '/settings');
+    });
+    it('Should link to profile', () => {
```
Comment (adrian-goe) [inline cypress/integration/Sidebar.spec.js:35]: lowercase

File: cypress/integration/auth.spec.js [hunk 28]
```diff
@@ -0,0 +1,16 @@
+describe('Login and registration test', () => {
```
Comment (adrian-goe) [inline cypress/integration/auth.spec.js:1]: lowercase

File: cypress/integration/auth.spec.js [hunk 29]
```diff
@@ -0,0 +1,16 @@
...
+    it('Should login', () => {
```
Comment (adrian-goe) [inline cypress/integration/auth.spec.js:2]: lowercase

File: cypress/integration/auth.spec.js [hunk 30]
```diff
@@ -0,0 +1,16 @@
...
+        cy.visit('/')
+            .get('.btn-login').click()
+            .url().should('include', '/dashboard');
+    });
+    it('Should register', () => {
```
Comment (adrian-goe) [inline cypress/integration/auth.spec.js:7]: lowercase

File: cypress/integration/auth.spec.js [hunk 31]
```diff
@@ -0,0 +1,16 @@
...
+        cy.visit('/')
+            .get('.btn-register').click()
+            .url().should('include', '/dashboard');
+    });
+    it('Should logout', () => {
```
Comment (adrian-goe) [inline cypress/integration/auth.spec.js:12]: lowercase

File: cypress/integration/footer.spec.js [hunk 32]
```diff
@@ -0,0 +1,17 @@
+describe('Footer links test', () => {
```
Comment (adrian-goe) [inline cypress/integration/footer.spec.js:1]: lowercase

File: cypress/integration/footer.spec.js [hunk 33]
```diff
@@ -0,0 +1,17 @@
...
+    it('Should show imprint', () => {
```
Comment (adrian-goe) [inline cypress/integration/footer.spec.js:2]: lowercase

File: cypress/integration/footer.spec.js [hunk 34]
```diff
@@ -0,0 +1,17 @@
...
+        cy.visit('/')
+            .get('.imprint-link').click()
+            .url().should('include', '/imprint');
+    });
+    it('Should show privacy', () => {
```
Comment (adrian-goe) [inline cypress/integration/footer.spec.js:7]: lowercase

File: cypress/integration/footer.spec.js [hunk 35]
```diff
@@ -0,0 +1,17 @@
...
+        cy.visit('/')
+            .get('.privacy-link').click()
+            .url().should('include', '/privacy');
+    });
+    it('Should show credits', () => {
```
Comment (adrian-goe) [inline cypress/integration/footer.spec.js:12]: lowercase

File: cypress/fixtures/example.json (hunk 2 above)
Comment (msentwickler) [reply cypress/fixtures/example.json:1]: No

**Pull request merged**